synapse/models/infotech.py

```diff
@@ -540,6 +627,180 @@ def getModelDefs(self):
+                    ('url', ('inet:url', {}), {
+                        'doc': 'A URL linking this CWE to a full description.',
+                    }),
+                    ('parents', ('array', {'type': 'it:sec:cwe', 'uniq': True, 'split': ','}), {
+                        'doc': 'An array of ChildOf CWE Relationships.'
+                    }),
+                )),
+                ('it:mitre:attack:group', {}, (
+                    ('org', ('ou:org', {}), {
+                        'doc': 'Used to map an ATT&CK group to a synapse ou:org.',
+                    }),
+                    ('name', ('ou:name', {}), {
+                        'doc': 'The primary name for the ATT&CK group.',
+                    }),
+                    ('names', ('array', {'type': 'ou:name', 'uniq': True}), {
+                        'doc': 'An array of alternate names for the ATT&CK group.',
+                    }),
+                    ('desc', ('str', {}), {
+                        'doc': 'A description of the ATT&CK group.',
+                        'disp': {'hint': 'text'},
+                    }),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'The URL that documents the ATT&CK group.',
+                    }),
+                    ('tag', ('syn:tag', {}), {
+                        'doc': 'The synapse tag used to annotate nodes included in this ATT&CK group ID.',
+                        'ex': 'cno.mitre.g0100',
+                    }),
+                    ('references', ('array', {'type': 'inet:url'}), {
+                        'doc': 'An array of URLs that document the ATT&CK group.',
+                    }),
+                    ('techniques', ('array', {'type': 'it:mitre:attack:technique', 'uniq': True, 'split': ','}), {
+                        'doc': 'An array of ATT&CK technique IDs used by the group.',
+                    }),
+                    ('software', ('array', {'type': 'it:mitre:attack:software', 'uniq': True, 'split': ','}), {
+                        'doc': 'An array of ATT&CK software IDs used by the group.',
+                    }),
+                )),
+                ('it:mitre:attack:tactic', {}, (
+                    ('name', ('str', {'strip': True}), {
+                        'doc': 'The primary name for the ATT&CK tactic.',
+                    }),
+                    ('desc', ('str', {}), {
+                        'doc': 'A description of the ATT&CK tactic.',
+                        'disp': {'hint': 'text'},
+                    }),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'The URL that documents the ATT&CK tactic.',
+                    }),
+                    ('tag', ('syn:tag', {}), {
+                        'doc': 'The synapse tag used to annotate nodes included in this ATT&CK tactic.',
+                        'ex': 'cno.mitre.ta0100',
+                    }),
+                    ('references', ('array', {'type': 'inet:url'}), {
```

Comment:
To fully model the ATT&CK model (IF we want to do this), then either a tactic needs an array of techniques that can be used with that tactic, or a technique needs an array of tactics that it can support. I am not sure we really want to do that (similar to proposed discussion about cross-ref'ing groups with software, etc.)

Reply:
Agreed.  Ideally we'd want to prevent overly "heavy" nodes and encode the "smaller subset of memberships on the larger population of things" so i'll add `:tactics` to `it:sec:mitre:attack:technique` :)